azbfs/url_service.go

```diff
@@ -1,25 +1,30 @@
 package azbfs
 
 import (
+	"context"
 	"net/url"
 
 	"github.com/Azure/azure-pipeline-go/pipeline"
 )
 
 // A ServiceURL represents a URL to the Azure Storage File service allowing you to manipulate file shares.
 type ServiceURL struct {
-	client managementClient
+	client filesystemClient
 }
 
 // NewServiceURL creates a ServiceURL object using the specified URL and request policy pipeline.
 func NewServiceURL(url url.URL, p pipeline.Pipeline) ServiceURL {
 	if p == nil {
 		panic("p can't be nil")
 	}
-	client := newManagementClient(url, p)
+	client := filesystemClient{newManagementClient(url, p)}
 	return ServiceURL{client: client}
 }
 
+func (s ServiceURL) ListFilesystemsSegment(ctx context.Context, marker *string) (*FilesystemList, error) {
```

Comment:
FileSystemURL is specific to a single filesystem. In my opinion, it would not make sense to have to specify a filesystem before we list other filesystems.

Reply:
I get it, that does make sense.